app/models/vote.rb

```diff
@@ -29,8 +29,12 @@ def self.noes
     where("votes.vote = 'no'")
   end
 
-  def rebellion?
-    !free? && vote != whip_guess
+  def rebellion?(whip = nil)
+    if whip.nil?
+      !free? && vote != whip_guess
+    else
+      !whip.free? && vote != whip.whip_guess
+    end
```

Comment:
The first `if` it uses delegate (as you said). When it call this method it does a query to get the whip and then the other queries related to that. The second `if` it gets from the `whip` from the parameter  (which is already loaded and then avoiding one extra query). 

It seems silly, but when in a for loop, that can avoid several queries to the DB. But it's ugly and I don't like it. If you have another suggestion to avoid this other query that would be good :smile: